CoreScriptsRoot/Modules/ChatSelector.lua

```diff
@@ -0,0 +1,184 @@
+			return mSignaler.Event:connect(function()
+				f(unpack(mArgData, 1, mArgDataCount))
+			end)
+		end
+
+		function sig:wait()
+			mSignaler.Event:wait()
+			assert(mArgData, "Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.")
+			return unpack(mArgData, 1, mArgDataCount)
+		end
+
+		return sig
+	end
+end
+
+
+
+local useModule = nil
+
+local state = {Visible = true}
+local interface = {}
+do
+	function interface:ToggleVisibility()
+		if (useModule) then
+			useModule:ToggleVisibility()
+		else
+			state.Visible = not state.Visible
+		end
+	end
+
+	function interface:SetVisible(visible)
+		if (useModule) then
+			useModule:SetVisible(visible)
+		else
+			state.Visible = visible
+		end
+	end
+
+	function interface:FocusChatBar()
+		if (useModule) then
+			useModule:FocusChatBar()
+		else
+			--// do nothing
+		end
+	end
+
+	function interface:GetVisibility()
+		if (useModule) then
+			return useModule:GetVisibility()
+		else
+			return state.Visible
+		end
+	end
+
+	function interface:GetMessageCount()
+		if (useModule) then
+			return useModule:GetMessageCount()
+		else
+			return 0
+		end
+	end
+
+	function interface:TopbarEnabledChanged(...)
+		if (useModule) then
+			return useModule:TopbarEnabledChanged(...)
+		else
+			
+		end
+	end
+
+	function interface:IsFocused(useWasFocused)
+		if (useModule) then
+			return useModule:IsFocused(useWasFocused)
+		else
+			return false
+		end
+	end
+
+	interface.ChatBarFocusChanged = Util.Signal()
+	interface.VisibilityStateChanged = Util.Signal()
+	interface.MessagesChanged = Util.Signal()
+end
+
+local stopCachingMakeSystemMessage = false
+local MakeSystemMessageCache = {}
+local function MakeSystemMessageCachingFunction(data)
+	if (stopCachingMakeSystemMessage) then return end
+	table.insert(MakeSystemMessageCache, data)
+end
+
+local function NonFunc() end
+StarterGui:RegisterSetCore("ChatMakeSystemMessage", MakeSystemMessageCachingFunction)
+StarterGui:RegisterSetCore("ChatWindowPosition", NonFunc)
+StarterGui:RegisterSetCore("ChatWindowSize", NonFunc)
+StarterGui:RegisterGetCore("ChatWindowPosition", NonFunc)
+StarterGui:RegisterGetCore("ChatWindowSize", NonFunc)
+StarterGui:RegisterSetCore("ChatBarDisabled", NonFunc)
+StarterGui:RegisterGetCore("ChatBarDisabled", NonFunc)
+
+
+local function ConnectSignals(useModule, interface, sigName)
+	useModule[sigName]:connect(function(...) interface[sigName]:fire(...) end)
+end
+
+if (TryLoadNewChat or FORCE_TRY_LOAD_NEW_CHAT) then
+	spawn(function()
+		local useNewChat = GetUseLuaFlag()
+		local useModuleScript = (useNewChat or FORCE_USE_NEW_CHAT) and RobloxGui.Modules.NewChat or RobloxGui.Modules.Chat
+		useModule = require(useModuleScript)
+
+		ConnectSignals(useModule, interface, "ChatBarFocusChanged")
+		ConnectSignals(useModule, interface, "VisibilityStateChanged")
+		ConnectSignals(useModule, interface, "MessagesChanged")
+
+		StarterGui:RegisterGetCore("UseNewLuaChat", function() return useNewChat end)
```

Comment:
No existing pattern (or existing, event would require a new RegisterCore functionality.
Simplified the way it is.